app/models/user.rb

```diff
@@ -289,11 +288,9 @@ def contact_for_school
   end
 
   def after_confirmation
-    return unless school.present?
+    return unless school&.visible
 
-    OnboardingMailer.with_user_locales(users: [self], school:) do |mailer|
-      mailer.welcome_email.deliver_now
-    end
+    OnboardingMailer.mailer.with(user: self, school:, locale: preferred_locale).welcome_email.deliver_now
```

Comment:
better to use deliver_later?

Reply:
Yes, would avoid doing the mailgun API call during the user model update. So worth changing